base/src/com/thoughtworks/go/util/OperatingSystem.java

```diff
@@ -12,44 +12,65 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- *************************GO-LICENSE-END***********************************/
+ */
 
 package com.thoughtworks.go.util;
 
-/**
- * @understands operating system type
-*/
-public enum OperatingSystem {
-    LINUX("Linux"), OSX("Mac OS X"), SUN_OS("SunOS"), WINDOWS("Windows"), UNKNOWN("Unknown");
-    private String name;
+import org.apache.commons.lang.StringUtils;
 
-    OperatingSystem(String name) {
-        this.name = name;
+import java.io.File;
+import java.io.FileReader;
+import java.util.Properties;
+import java.util.Scanner;
+
+public class OperatingSystem {
+    private static final String OS_FAMILY_NAME = System.getProperty("os.name");
+    public static final String WINDOWS = "Windows";
+    private static String OS_COMPLETE_NAME = detectCompleteName();
+
+    private static String detectCompleteName() {
+        String[] command = {"python", "-c", "import platform;print(platform.linux_distribution())"};
+        try {
+            Process process = Runtime.getRuntime().exec(command);
```

Comment:
So, we have something called ```CommandLine#run``` and ```CommandLine#runOrBomb``` to execute commands. May be we could use that? @ketan, what do you think?